aws-synthetics-group/src/main/java/com/amazon/synthetics/group/UpdateHandler.java

```diff
@@ -0,0 +1,145 @@
+import software.amazon.cloudformation.Action;
+import software.amazon.cloudformation.exceptions.CfnGeneralServiceException;
+import software.amazon.cloudformation.exceptions.CfnResourceConflictException;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.OperationStatus;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+
+public class UpdateHandler extends BaseHandlerStd {
+    private Logger logger;
+
+    public UpdateHandler() {
+        super(Action.UPDATE);
+    }
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest() {
+        try {
+            if (!callbackContext.isGroupUpdateStarted()) {
+                // We just need the group at the start to get current arn list, we will rely on exceptions from AssociateResource to handle concurrent mods
+                // This will need to be revisited when we increase resource limit
+                log("Started update request");
+                Group group = getGroupOrThrow();
+                List<String> groupResources = getGroupResourcesOrThrow();
+                diffGroupArnList(groupResources);
+                if (model.getTags() != null) {
+                    Map<String, Map<String, String>> tagResourceMap = TagHelper.updateTags(model, group.tags());
+                    String groupArn = TagHelper.buildGroupArn(request, group.id());
+                    if (!tagResourceMap.get(Constants.ADD_TAGS).isEmpty()) {
+                        addTags(tagResourceMap, groupArn);
+                    }
+
+                    if (!tagResourceMap.get(Constants.REMOVE_TAGS).isEmpty()) {
+                        removeTags(tagResourceMap, groupArn);
+                    }
+                }
+                callbackContext.setGroupUpdateStarted(true);
+                return ProgressEvent.<ResourceModel, CallbackContext>builder()
+                    .resourceModel(model)
+                    .callbackContext(callbackContext)
+                    .callbackDelaySeconds(Constants.DEFAULT_CALLBACK_DELAY_SECONDS)
+                    .message(Constants.GROUP_UPDATE_DIFF_COMPLETE)
+                    .status(OperationStatus.IN_PROGRESS)
+                    .build();
+            }
+
+            if (callbackContext.getAddResourceListIndex() == callbackContext.getAddResourceList().size() &&
+                callbackContext.getRemoveResourceListIndex() == callbackContext.getRemoveResourceList().size()) {
+                return ProgressEvent.defaultSuccessHandler(model);
+            }
+
+            if (!callbackContext.isGroupAssociationStarted()) {
+                callbackContext.setGroupAssociationStarted(true);
+            }
+
+            if (callbackContext.isGroupAssociationStarted() &&
+                callbackContext.getAddResourceListIndex() < callbackContext.getAddResourceList().size()) {
+                return addAssociatedResources(true);
```

Comment:
The resource list can contain canaries from different regions. Associate/Disassociate apis require the client to call from canary region otherwise it will throw ValidationException. Can't tell if you are handling it correctly, just flagging it here.

Reply:
That's a big one.. not handling this now. I will make the changes to support it